explorer/main.cpp

```diff
@@ -70,13 +70,17 @@ static auto Main(llvm::StringRef default_prelude_file, int argc, char* argv[])
   }
 
   Arena arena;
-  CARBON_ASSIGN_OR_RETURN(AST ast,
-                          Parse(&arena, input_file_name, parser_debug));
+  CARBON_ASSIGN_OR_RETURN(
+      AST ast,
+      Parse(&arena, input_file_name, parser_debug).WithPrefix("SYNTAX ERROR"));
```

Comment:
```suggestion
      Parse(&arena, input_file_name, parser_debug).WithPrefix("PARSE ERROR"));
```

Maybe slightly different?

Reply:
I started with that, but this also diagnoses lexical errors, which are syntax problems but not parsing problems, so I think "syntax error" is more accurate.